server/resp/src/test/java/org/infinispan/server/resp/types/TransactionMediaTypeTest.java

```diff
@@ -0,0 +1,109 @@
+
+@Test(groups = "functional", testName = "server.resp.types.TransactionMediaTypeTest")
+public class TransactionMediaTypeTest extends TransactionOperationsTest {
+
+   private CacheMode cacheMode;
+   private boolean simpleCache;
+   private MediaType valueType;
+
+   protected EmbeddedCacheManager createTestCacheManager() {
+      GlobalConfigurationBuilder globalBuilder = new GlobalConfigurationBuilder();
+      globalBuilder = cacheMode == CacheMode.LOCAL
+            ? globalBuilder.nonClusteredDefault()
+            : globalBuilder.clusteredDefault();
+      TestCacheManagerFactory.amendGlobalConfiguration(globalBuilder, new TransportFlags());
+      ConfigurationBuilder builder = new ConfigurationBuilder();
+      builder.encoding().key().mediaType(MediaType.APPLICATION_OCTET_STREAM);
+      builder.clustering().hash().keyPartitioner(new CRC16HashFunctionPartitioner()).numSegments(256);
+      amendConfiguration(builder);
+      EmbeddedCacheManager cacheManager = TestCacheManagerFactory.newDefaultCacheManager(true, globalBuilder, builder);
+      TestingUtil.replaceComponent(cacheManager, TimeService.class, timeService, true);
+      return cacheManager;
+   }
+
+   @Override
+   protected EmbeddedCacheManager createCacheManager() {
+      cacheManager = createTestCacheManager();
+      Configuration configuration = new ConfigurationBuilder()
+            .encoding()
+            .key().mediaType(MediaType.APPLICATION_OCTET_STREAM)
+            .encoding()
+            .value().mediaType(valueType.toString())
+            .build();
+      RespServerConfiguration serverConfiguration = serverConfiguration().build();
+      cacheManager.defineConfiguration(serverConfiguration.defaultCacheName(), configuration);
+      server = startServer(cacheManager, serverConfiguration);
+      client = createClient(30000, server.getPort());
+      redisConnection = client.connect();
+      cache = cacheManager.getCache(server.getConfiguration().defaultCacheName());
+      return cacheManager;
+   }
+
+   @Override
+   protected void amendConfiguration(ConfigurationBuilder configurationBuilder) {
+      if (simpleCache) {
+         configurationBuilder.clustering().cacheMode(CacheMode.LOCAL).simpleCache(true);
+      } else {
+         configurationBuilder.clustering().cacheMode(cacheMode);
+      }
+   }
+
+   private TransactionMediaTypeTest withValueType(MediaType type) {
+      this.valueType = type;
+      return this;
+   }
+
+   private TransactionMediaTypeTest withSimpleCache() {
+      this.simpleCache = true;
+      return this;
+   }
+
+   private TransactionMediaTypeTest withCacheMode(CacheMode mode) {
+      this.cacheMode = mode;
+      return this;
+   }
+
+   @Factory
+   public Object[] factory() {
+      List<TransactionMediaTypeTest> instances = new ArrayList<>();
+      MediaType[] types = new MediaType[] {
+            MediaType.APPLICATION_OCTET_STREAM,
+            MediaType.APPLICATION_OBJECT,
+            MediaType.TEXT_PLAIN,
```

Comment:
Lets add PROTOSTREAM to the various media type tests as well just to be safe.

Reply:
Nice. This one found a little problem with the pub sub. I changed to not use the data conversion while handling the event. Instead, we use a converter.